sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -228,24 +355,36 @@ class AccessPolicy(GenAccessPolicy):
         be interpreted as UTC.
     :type start: ~datetime.datetime or str
     """
-    def __init__(self, permission=None, expiry=None, start=None):
+
+    permission: Optional[Union[ShareSasPermissions, str]]  # type: ignore [assignment]
```

Comment:
Yeah this is a bit strange. Here's what `mypy` says after removing `# type: ignore [assignment]`:
```
azure/storage/fileshare/_models.py:360: error: Incompatible types in assignment (expression has type "Union[ShareSasPermissions, str, None]", base class "AccessPolicy" defined the type as "Optional[str]")  [assignment]
azure/storage/fileshare/_models.py:363: error: Incompatible types in assignment (expression has type "Union[datetime, str, None]", base class "AccessPolicy" defined the type as "Optional[str]")  [assignment]
azure/storage/fileshare/_models.py:365: error: Incompatible types in assignment (expression has type "Union[datetime, str, None]", base class "AccessPolicy" defined the type as "Optional[str]")  [assignment]
```

Not sure why the base class `AccessPolicy` pops up since we're assigning params directly to instance variables.

Reply:
In addition, `mypy` won't accept `Optional[...] = None` as a way to resolve the errors above.

Reply:
Looks like this comes from `GeneratedAccessPolicy`.